File: android/app/src/main/java/com/preuvely/app/data/models/Claim.kt
package com.preuvely.app.data.models

import com.google.gson.annotations.SerializedName
import java.text.SimpleDateFormat
import java.util.*

enum class ClaimStatus(val value: String) {
    @SerializedName("pending") PENDING("pending"),
    @SerializedName("approved") APPROVED("approved"),
    @SerializedName("rejected") REJECTED("rejected")
}

data class Claim(
    @SerializedName("id") val id: Int,
    @SerializedName("store_id") val storeId: Int,
    @SerializedName("store_name") val storeName: String?,
    @SerializedName("store_slug") val storeSlug: String?,
    @SerializedName("requester_name") val requesterName: String,
    @SerializedName("requester_phone") val requesterPhone: String,
    @SerializedName("note") val note: String?,
    @SerializedName("status") val status: ClaimStatus,
    @SerializedName("reject_reason") val rejectReason: String?,
    @SerializedName("created_at") val createdAt: String?
) {
    val formattedDate: String
        get() {
            createdAt ?: return ""
            return try {
                val inputFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.getDefault())
                val outputFormat = SimpleDateFormat("MMM d, yyyy", Locale.getDefault())
                val date = inputFormat.parse(createdAt)
                date?.let { outputFormat.format(it) } ?: ""
            } catch (e: Exception) {
                ""
            }
        }

    val displayStoreName: String
        get() = storeName ?: "Unknown Store"
}

data class ClaimListResponse(
    @SerializedName("data") val data: List<Claim>
)

data class ClaimResponse(
    @SerializedName("message") val message: String,
    @SerializedName("data") val data: Claim
)

File: ios/Preuvely/Preuvely/Models/Claim.swift
import Foundation
import Combine

// MARK: - Claim Status

enum ClaimStatus: String, Codable, Hashable {
    case pending
    case approved
    case rejected

    var displayName: String {
        switch self {
        case .pending: return "Pending"
        case .approved: return "Approved"
        case .rejected: return "Rejected"
        }
    }

    var localizedKey: String {
        switch self {
        case .pending: return L10n.Claim.pending
        case .approved: return L10n.Claim.approved
        case .rejected: return L10n.Claim.rejected
        }
    }
}

// MARK: - Claim

struct Claim: Identifiable, Codable, Hashable {
    let id: Int
    let storeId: Int
    let storeName: String?
    let storeSlug: String?
    let requesterName: String
    let requesterPhone: String
    let note: String?
    let status: ClaimStatus
    let rejectReason: String?
    let createdAt: Date?

    // Note: Using JSONDecoder with .convertFromSnakeCase strategy

    var formattedDate: String {
        guard let date = createdAt else { return "" }
        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        formatter.timeStyle = .none
        return formatter.string(from: date)
    }

    /// Display name for the store (falls back to store ID if name not available)
    var displayStoreName: String {
        storeName ?? "Store #\(storeId)"
    }
}

// MARK: - Preview Data

#if DEBUG
extension Claim {
    static let samples: [Claim] = []
    static let sample = Claim(
        id: 1,
        storeId: 1,
        storeName: "Preview Store",
        storeSlug: "preview-store",
        requesterName: "Preview User",
        requesterPhone: "+213000000000",
        note: nil,
        status: .pending,
        rejectReason: nil,
        createdAt: Date()
    )
}
#endif
